text/0100-sensitive-data-handling.md

```diff
@@ -0,0 +1,103 @@
+#  Sensitive Data Handling
+
+By default, OpenTelemetry libraries never capture potentially-sensitive data, except for the full URL.
+
+## Motivation
+
+Many businesses are under regulatory or contractual constraints on how they process sensitive data (e.g., GDPR, 
+PCI, HIPPA).  If we are "safe by default" and manage this issue well, it will help greatly with 
+OpenTelemetry adoption in many markets.
+
+## Explanation
+
+By default, our java auto-instrumentation library does not capture your app's credit card numbers, taxpayer IDs, 
+personal medical data, users' passwords, etc.  Unless you put them in the URL.  If you have put them in the URL, we provide a way to fix that.
+
+How do we do that?  For example, we don't know the semantics of the SQL your app sends to its database(s).  Some of the data might be 
+sensitive, and some of it might not.  What we do is remove all 
+*potentially* sensitive values.  This means that sql like `SELECT * FROM USERS WHERE ID=123456789` gets turned into
+`SELECT * FROM USERS WHERE ID=?` when it gets put into our `db.statement` attribute.  Since our library doesn't know anything about the 
+meaning of `ID` values, we decided to be "safe by default" and drop all numeric and string values entirely from all SQL.  So what gets 
+captured in our system is just table names, field names, clause structures, etc. (metadata about your data's structure and how you access it).
+
+Yes, this also has the pleasant side effect of reducing the cardinality of these dimensions, making it easier to build metrics or indices on them. 
+If you want to reduce it further, our collector has features for that.  Again, our design goal is to protect you from 
+liability or exposure due to captured sensitive data.
+
+We use similar techniques for all other protocols/query/command mechanisms.  One exception is how we treat URLs.  We will by
+default capture the full URL (including the query and fragment) in `http.url`.  Note that this will not include
+the contents of form submissions or XHR posts.  Since many products/features log the entire URL already 
+(like in Apache's access log), developers already need to consider this.  If you are putting sensitive data in the URL,
+we offer configuration options to work around this.
+
+If your app doesn't process any sensitive data at all and you want to see the raw sql values, you can configure that.  The documentation
+will have some bold/red warnings telling you to be sure you know what you're doing.
```

Comment:
Is this something to be configured in the UI of a backend or at the place where the instrumentation is set up? Are these bold red warnings supposed to show up in a log produced by the instrumentation or in the tracing backend's UI?

Reply:
Config would remain local to each library (a manual wrapper might provide a dictionary of options as an API parameter, an auto agent might use an environment variable or config file).  The warning would appear in the documentation of this config knob.